components/flow/flow-service/src/main/kotlin/net/corda/flow/application/services/impl/interop/binding/FacadeInterfaceBinding.kt

```diff
@@ -0,0 +1,137 @@
+package net.corda.flow.application.services.impl.interop.binding
+
+import net.corda.v5.application.interop.facade.Facade
+import net.corda.v5.application.interop.facade.FacadeMethod
+import net.corda.v5.application.interop.parameters.TypedParameter
+import java.lang.reflect.Constructor
+import java.lang.reflect.Method
+import kotlin.reflect.KFunction
+import kotlin.reflect.jvm.javaMethod
+
+
+/**
+ * Represents the binding of a [Facade] to a JVM interface.
+ *
+ * @param facade The bound [Facade]
+ * @param boundInterface The bound JVM interface.
+ * @param methodBindings Bindings of Facade methods to interface methods.
+ */
+data class FacadeInterfaceBinding(
+    val facade: Facade,
+    val boundInterface: Class<*>,
+    val methodBindings: List<FacadeMethodBinding>
+) {
+
+    private val bindingsByMethod = methodBindings.associateBy { it.interfaceMethod }
+    private val bindingsByFacadeMethodName = methodBindings.associateBy { it.facadeMethod.name }
+
+    /**
+     * Obtain the [FacadeMethodBinding] for a given [Method] on the bound interface.
+     *
+     * @param interfaceMethod The [Method] to get the binding for
+     */
+    fun bindingFor(interfaceMethod: Method): FacadeMethodBinding? = bindingsByMethod[interfaceMethod]
+
+    /**
+     * Utility method for writing tests in Kotlin - enables us to request the binding for a method by passing in a
+     * Kotlin method reference, e.g. `facadeBindings.bindingFor(MyInterface::myMethod)`.
+     *
+     * @param method A Kotlin method reference pointing to the method to gether the binding for
+     */
+    fun bindingFor(method: KFunction<*>) = bindingFor(method.javaMethod!!)
+    fun bindingFor(facadeMethodName: String): FacadeMethodBinding? = bindingsByFacadeMethodName[facadeMethodName]
+}
+
+/**
+ * Binding of a [FacadeMethod] to a [Method] on a JVM interface.
+ *
+ * @param facadeMethod The bound [FacadeMethod]
+ * @param interfaceMethod The bound JVM [Method]
+ * @param inParameterBindings Bindings of method parameters to [TypedParameter] in-parameters
+ * @param outParameterBindings Bindings of [TypedParameter] out-parameters, either to a single value or to fields in a
+ * data class
+ */
+data class FacadeMethodBinding(
+    val facadeMethod: FacadeMethod,
+    val interfaceMethod: Method,
+    val inParameterBindings: List<FacadeInParameterBinding>,
+    val outParameterBindings: FacadeOutParameterBindings
+) {
+
+    private val inParameterBindingsByName = inParameterBindings.associateBy { it.facadeParameter.name }
+
+    /**
+     * Obtain the [FacadeInParameterBinding] for the method parameter at the given index
+     *
+     * @param parameterIndex The index of the method parameter to get the binding for
+     */
```

Comment:
```suggestion
    /**
     * Obtain the [FacadeInParameterBinding] for the method parameter at the given index
     * @param parameterIndex The index of the method parameter to get the binding for
     */
```

Please change in other places